src/Runtime/ShutdownHandler.php

```diff
@@ -42,6 +42,11 @@ public function shutdownHandler()
 
         if (!Drush::config()->get(Runtime::DRUSH_RUNTIME_COMPLETED_NAMESPACE)) {
             Drush::logger()->warning('Drush command terminated abnormally. Check for an exit() in your Drupal site.');
+            // Make sure that we will return an error code when we exit,
+            // even if the code that got us here did not.
+            if (!Runtime::exitCode()) {
```

Comment:
Should we unmark this method as deprecated?

Reply:
I don't think so. I'd like to eventually rework exit code handling. Really, the `Runtime` class should be considered internal to Drush. We should put APIs that we want folks to use in the Drush class.